File: frontend/src/features/properties/types/PropertyTrace.ts
import type Property from "./Property";

export interface PropertyTrace {
  id: string;
  name: string;
  dateSale: Date | null;
  value: number | null;
  tax: number | null;
  idProperty: string | undefined;
};

export interface PropertyWithTraces {
    property: Property;
    propertyTraces: PropertyTrace[]
}

File: backend/RealEstate.Domain/Entities/PropertyTrace.cs
using MongoDB.Bson.Serialization.Attributes;
using MongoDB.Bson;

namespace RealEstate.Domain.Entities
{
    public class PropertyTrace
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string Id { get; set; } = string.Empty;

        [BsonElement("dateSale")]
        public DateOnly DateSale { get; set; }

        [BsonElement("name")]
        public string Name { get; set; } = string.Empty;

        [BsonElement("value")]
        public decimal value { get; set; }

        [BsonElement("tax")]
        public decimal Tax { get; set; }

        [BsonElement("idProperty")]
        public string IdProperty { get; set; } = string.Empty;
    }
}
